checks/pinned_dependencies_test.go

```diff
@@ -196,6 +196,58 @@ func TestNonGithubWorkflowPinning(t *testing.T) {
 	}
 }
 
+func TestGithubWorkflowPkgManagerPinning(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		filename string
+		expected scut.TestReturn
+	}{
+		{
+			name:     "npm packages without verification",
+			filename: "./testdata/github-workflow-pkg-managers.yaml",
+			expected: scut.TestReturn{
+				Error:         nil,
+				Score:         checker.MinResultScore,
+				NumberOfWarn:  4,
+				NumberOfInfo:  0,
+				NumberOfDebug: 0,
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var content []byte
+			var err error
+
+			content, err = ioutil.ReadFile(tt.filename)
+			if err != nil {
+				panic(fmt.Errorf("cannot read file: %w", err))
+			}
+
+			dl := scut.TestDetailLogger{}
+			s, e := testValidateGitHubWorkflowScriptFreeOfInsecureDownloads(tt.filename, content, &dl)
+			actual := checker.CheckResult{
+				Score:  s,
+				Error2: e,
+			}
+
+			// TODO: Validate offsets
+			// expectedOffsets := []int{38, 40, 42, 44}
```

Comment:
will u do this in a follow-up PR?
Create a different test specifically for it.

Reply:
Will fix this in a follow up PR. We'll have to create a mechanism to track line numbers per file type since the AST parser doesn't start at line 1 for non shell script (`.sh`) files.